server/src/main/java/com/epam/aidial/core/server/service/ApplicationService.java

```diff
@@ -417,7 +421,11 @@ public Application.Logs getApplicationLogs(ResourceDescriptor resource) {
     private void prepareApplication(ResourceDescriptor resource, Application application) {
         verifyApplication(resource);
 
-        if (application.getEndpoint() == null && application.getFunction() == null) {
+        if (application.getCustomAppSchemaId() != null) {
+            if (application.getEndpoint() != null) {
```

Comment:
Please add check that function is null (if not throw exception)

Reply:
done